### PR TITLE
[MRG]Fix model retrieving from modelhub

### LINKED ISSUE #287: Add model retrieving from modelhub support
#### Software and Hardware Versions 
ModelCI v1.x.x, CUDA Version vx.x.x, GPU device used...

#### Problem description

Currently, ModelCI saves model object into GridFS, but don't support file retrieving currently

#### Other Information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by univerone**
Closes #287

#### What has been done to verify that this works as intended?

Now model object can be retrieved and loaded by `saved_path`:

```python
model = get_by_id("6075bc371f60af4e3bf64bc4")
torch_model = torch.load(model.saved_path)
```

or by reading from GridFS

```python
  model = get_by_id("6075bc371f60af4e3bf64bc4")
  buffer = io.BytesIO(model.weight.__bytes__())
  torch_model = torch.load(buffer)
```

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: modelci/types/models/mlmodel.py [hunk 1]
```diff
@@ -7,45 +7,61 @@
...
         if isinstance(__root__, Path):
             object.__setattr__(self, 'file', FilePath.validate(__root__))
+            object.__setattr__(self, 'grid_out', None)
             __root__ = None
 
-        self._grid_out: Optional[GridOut]
+        if isinstance(__root__, ObjectId):
+            if _fs.exists(__root__):
+                object.__setattr__(self, 'file', None)
+                object.__setattr__(self, 'grid_out', _fs.get(__root__))
```
Comment (YuanmingLeee) [inline modelci/types/models/mlmodel.py:45]: What does `grid_out` means?

Comment (YuanmingLeee) [reply modelci/types/models/mlmodel.py:45]: We should hide GridFS knowledge from the user side.

Comment (univerone) [reply modelci/types/models/mlmodel.py:45]: `Weight` class has a `_grid_out` properties previously, it's GridOut instance of target model file

Comment (univerone) [reply modelci/types/models/mlmodel.py:45]: > We should hide GridFS knowledge from the user side.

Should we make `grid_out` private attribute?

Comment (HuaizhengZhang) [reply modelci/types/models/mlmodel.py]: better gridfs_out as grid is misleading.

**Review by HuaizhengZhang — APPROVED** [review]
Thanks for the PR. Revise accordingly and then merge

**Review by HuaizhengZhang — APPROVED** [review]

**Pull request merged**